can/zynqmp-can/zynqmp-can.c

```diff
@@ -0,0 +1,552 @@
+	const uint32_t ECR;       /* Rx and Tx Error Counters, Address offset: 0x10 */
+	uint32_t ESR;             /* Error Status, Address offset: 0x14 */
+	uint32_t SR;              /* Controller Status, Address offset: 0x18 */
+	uint32_t ISR;             /* Interrupt Status, Address offset: 0x1c */
+	uint32_t IER;             /* Interrupt Enable, Address offset: 0x20 */
+	uint32_t ICR;             /* Interrupt Clear, Address offset: 0x24 */
+	uint32_t TCR;             /* Timestamp Clear., Address offset: 0x28 */
+	uint32_t WIR;             /* Rx and Tx Watermark Settings., Address offset: 0x2c */
+	uint32_t TXFIFO_ID;       /* Tx Message FIFO, Identifier, Request., Address offset: 0x30 */
+	uint32_t TXFIFO_DLC;      /* Tx Message FIFO Data Length Code., Address offset: 0x34 */
+	uint32_t TXFIFO_DATA1;    /* Tx Message FIFO, data word 1., Address offset: 0x38 */
+	uint32_t TXFIFO_DATA2;    /* Tx Message FIFO, data word 2., Address offset: 0x3c */
+	uint32_t TXHPB_ID;        /* High Priority Tx Message FIFO, Identifier, Request., Address offset: 0x40 */
+	uint32_t TXHPB_DLC;       /* High Priority Tx Message FIFO Data Length Code., Address offset: 0x44 */
+	uint32_t TXHPB_DATA1;     /* High Priority Tx Message FIFO, data word 1., Address offset: 0x48 */
+	uint32_t TXHPB_DATA2;     /* High Priority Tx Message FIFO, data word 0., Address offset: 0x4c */
+	const uint32_t RXFIFO_ID; /* Rx Message FIFO, Identifier, Request., Address offset: 0x50 */
+	uint32_t RXFIFO_DLC;      /* Rx Message FIFO Data Length Code., Address offset: 0x54 */
+	uint32_t RXFIFO_DATA1;    /* Rx Message FIFO, data word 1., Address offset: 0x58 */
+	uint32_t RXFIFO_DATA2;    /* Rx Message FIFO, data word 2., Address offset: 0x5c */
+	uint32_t AFR;             /* Acceptance Filter Enables., Address offset: 0x60 */
+	uint32_t AFMR1;           /* Acceptance Filter 1 Mask., Address offset: 0x64 */
+	uint32_t AFIR1;           /* Acceptance Filter 1 ID., Address offset: 0x68 */
+	uint32_t AFMR2;           /* Acceptance Filter 2 Mask., Address offset: 0x6c */
+	uint32_t AFIR2;           /* Acceptance Filter 2 ID., Address offset: 0x70 */
+	uint32_t AFMR3;           /* Acceptance Filter 3 Mask., Address offset: 0x74 */
+	uint32_t AFIR3;           /* Acceptance Filter 3 ID., Address offset: 0x78 */
+	uint32_t AFMR4;           /* Acceptance Filter 4 Mask., Address offset: 0x7c */
+	uint32_t AFIR4;           /* Acceptance Filter 4 ID., Address offset: 0x80 */
+} can_periph_t;
+
+/* Data structure used by the driver */
+static struct {
+	volatile can_periph_t *base; /**< Pointer to the peripheral base address */
+	oid_t oid;                   /**< Object identifier for the related device file */
+	handle_t cond;               /**< Conditional variable for synchronizing interrupts */
+	handle_t inth;               /**< Interrupt handler object */
+	handle_t lock;               /**< Mutex used with the conditional variable for synchronization */
+} can = { 0 };
```

Comment:
Initialized is redundant
```suggestion
} can;
```